### PR TITLE
Normalise csm after upsampling

### LINKED ISSUE #797: Lowres CSMs result in artifacts due to upsampling
Low-Res CSMs result in artifacts in the reconstruction
Image

If using linear upsampling, we need to normalize afterwards: 

consider two coils at  two neighbouring pixels
0|1  and 1|0:
if we add a new interpolated pixel in between, it will be .5|.5 -- which will not be normalized

Alternativly, changing the interpolation mode to nearest would also fix it.

### PR DESCRIPTION
**Pull request by ckolbPTB**
Closes #797

### CODE REVIEW

File: src/mrpro/data/CsmData.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ def from_idata_walsh(
             chunk_size=chunk_size_otherdim,
         )
         csm_tensor = csm_fun(idata.data.flatten(end_dim=-5)).reshape(idata.data.shape)
+        # upsampled csm requires normalization
+        csm_tensor /= csm_tensor.abs().square().sum(dim=-4, keepdim=True).sqrt() + 1e-9
```
Comment (fzimmermann89) [inline src/mrpro/data/CsmData.py:139]: csm_tensor = torch.nn.functional.normalize(csm_tensor, p=2, dim=-4, eps=1e-9)

Comment (fzimmermann89) [inline src/mrpro/data/CsmData.py]:
```suggestion
        csm_tensor = torch.nn.functional.normalize(csm_tensor, p=2, dim=-4, eps=1e-9)
```

File: src/mrpro/data/CsmData.py [hunk 2]
```diff
@@ -221,6 +223,8 @@ def from_idata_inati(
...
```
Comment (fzimmermann89) [inline src/mrpro/data/CsmData.py]:
```suggestion
        csm_tensor = torch.nn.functional.normalize(csm_tensor, p=2, dim=-4, eps=1e-9)
```

**Review by fzimmermann89 — APPROVED** [review]
just in case we ever want to autograd here, should already avoid inplace if possible.

also, you could consider swithcing to F.normalize (see suggestions).
